sigmastate/src/main/scala/sigmastate/interpreter/ProverInterpreter.scala

```diff
@@ -156,19 +177,38 @@ trait ProverInterpreter extends Interpreter with ProverUtils with AttributionCor
       error(s"Don't know how to markReal($t)")
   })
 
+  /**
+    * Set positions for children of a unproven inner node (conjecture, so AND/OR/THRESHOLD)
+    */
+  private def setPositions(uc: UnprovenConjecture): UnprovenConjecture = {
+    val updChildren = uc.children.zipWithIndex.map { case (pt, idx) =>
+        pt.asInstanceOf[UnprovenTree].withPosition(uc.position + "-" + idx.toString)
+    }
+    uc match {
+      case and: CAndUnproven => and.copy(children = updChildren)
+      case or: COrUnproven => or.copy(children = updChildren)
+      case threshold: CThresholdUnproven => threshold.copy(children = updChildren)
+    }
+  }
```

Comment:
Please add test-vector-like tests for this method, this will also help see how the positions are assigned.
This is to catch regressions, because applications will rely on exact behavior of this code.

Reply:
Please also base this PR on top of #677

Reply:
rebased

Reply:
test done , see ProverSpecification

Reply:
The method is not covered by the test
<img width="1054" alt="image" src="https://user-images.githubusercontent.com/654077/93190207-bdf85d80-f74b-11ea-8fc9-a6b7090197d6.png">

Reply:
made new tests for this